### ISSUE TITLE
anyone experienced `rh.get_all_open_option_orders(account_number=acct)` is very slow? e.g. ~2 minutes

### ISSUE THREAD

**Issue by mw66** [jmfernandes/robin_stocks#507]
Hi,

Just wonder if anyone experienced `rh.get_all_open_option_orders(account_number=acct)` is very slow? e.g. ~2 minutes

Is this a server problem, or anything we can improve it on the client side code?

Thanks.

**Comment by noLeash** [comment]
@mw66 Yes, I was having this issue and have resolved it. The problem is that it's requesting the entire list of transactions for the account. I've started a fork that solves this issue by letting you send a start datetime for how far back to request records, and I will post the update tomorrow.

**Comment by noLeash** [comment]
@mw66 - Fork here https://github.com/noLeash/robin_stocks/tree/order-req-start-date

**Comment by mw66** [comment]
@noLeash Thanks for the information and the fork, can you make a PR and contribute back to the original repo?

@jmfernandes

**Comment by noLeash** [comment]
@mw66 - PR has been submitted https://github.com/jmfernandes/robin_stocks/pull/517

**Comment by OnishiKenshin** [comment]
@noLeash hey, just wanted to comment a thanks for your fix on this.

**Comment by noLeash** [comment]
> @noLeash hey, just wanted to comment a thanks for your fix on this.

Happy to be able to contribute
